lib/Crypto/importkeyrequest.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef LIBSAILFISHCRYPTO_IMPORTKEYREQUEST_H
+#define LIBSAILFISHCRYPTO_IMPORTKEYREQUEST_H
+
+#include "Crypto/cryptoglobal.h"
+#include "Crypto/request.h"
+#include "Crypto/key.h"
+#include "Crypto/interactionparameters.h"
+
+#include <QtCore/QObject>
+#include <QtCore/QScopedPointer>
+#include <QtCore/QString>
+
+namespace Sailfish {
+
+namespace Crypto {
+
+class CryptoManager;
+
+class ImportKeyRequestPrivate;
+class SAILFISH_CRYPTO_API ImportKeyRequest : public Sailfish::Crypto::Request
+{
+    Q_OBJECT
+    Q_PROPERTY(QString cryptoPluginName READ cryptoPluginName WRITE setCryptoPluginName NOTIFY cryptoPluginNameChanged)
+    Q_PROPERTY(Sailfish::Crypto::InteractionParameters interactionParameters READ interactionParameters WRITE setInteractionParameters NOTIFY interactionParametersChanged)
+    Q_PROPERTY(Sailfish::Crypto::Key key READ key WRITE setKey NOTIFY keyChanged)
+    Q_PROPERTY(Sailfish::Crypto::Key importedKey READ importedKey NOTIFY importedKeyChanged)
+
+public:
+    ImportKeyRequest(QObject *parent = Q_NULLPTR);
+    ~ImportKeyRequest();
```

Comment:
The destructor is an override of QObject's destructor and Qt convention is to not add the virtual keyword to overrides. Nowdays it could have an override keyword instead though.

Reply:
Ah, ok. Then `Q_DECL_OVERRIDE` is needed here.